#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/BrushedAluminium.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/CementTiles.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/street_furniture/protos/Bench.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/BakelitePlastic.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.20346163355955785 -0.23927229615382126 -0.9493956667072767 4.713095385043199
  position -0.6282810094456429 -0.9733082580875537 1.1010728003633188
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 10 10
  wallThickness 0.1
  wallHeight 0.5
}
Robot {
  translation -0.59998 0.0248868 0.293629
  children [
    Solid {
      translation 0 0 0.2
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
            position 1.57
            axis 0 1 0
            anchor 0 0 0.05
          }
          device [
            RotationalMotor {
              name "camera motor"
            }
          ]
          endPoint Solid {
            translation 0.10000001585340829 0 0.04996018366446334
            rotation 0 1 0 1.57
            children [
              Camera {
                translation -7.13725e-07 0 0.16
                rotation 0 1 0 -1.5707953071795862
              }
              Shape {
                appearance BakelitePlastic {
                }
                geometry Box {
                  size 0.05 0.1 0.3
                }
              }
            ]
          }
        }
        Shape {
          appearance BakelitePlastic {
          }
          geometry Box {
            size 0.05 0.1 0.1
          }
        }
      ]
      name "Camera Joint"
    }
    TouchSensor {
      translation 0.25 0.1 0
      rotation 0 0 1 0.261799
      children [
        DEF TouchSensor Shape {
          appearance CementTiles {
          }
          geometry Box {
            size 0.05 0.15 0.15
          }
        }
      ]
      name "left sensor"
      boundingObject USE TouchSensor
    }
    TouchSensor {
      translation 0.25 -0.1 0
      rotation 0 0 1 -0.261799
      children [
        DEF TouchSensor Shape {
          appearance CementTiles {
          }
          geometry Box {
            size 0.05 0.15 0.15
          }
        }
      ]
      name "right sensor"
      boundingObject USE TouchSensor
    }
    Track {
      translation 0 0.3 -0.125
      children [
        TrackWheel {
          position 0.175 0
          rotation 0.2013943978978304 0.6926170558994189 -0.6926196000494003 2.744122141777874
          children [
            Shape {
              geometry Box {
                size 0.1 0.1 0.1
              }
            }
            DEF Sandwhich_Wheel Shape {
              appearance PBRAppearance {
              }
              geometry Cylinder {
                height 0.05
                radius 0.1
              }
            }
          ]
        }
        TrackWheel {
          position -0.175 0
          rotation 0.2013943978978304 0.6926170558994189 -0.6926196000494003 2.744122141777874
          children [
            Shape {
              geometry Box {
                size 0.1 0.1 0.1
              }
            }
            DEF Sandwhich_Wheel Shape {
              appearance PBRAppearance {
              }
              geometry Cylinder {
                height 0.05
                radius 0.1
              }
            }
          ]
        }
      ]
      name "track1"
      boundingObject Group {
        children [
          Pose {
            translation 0.175 0 0
            rotation 1 0 0 1.5708
            children [
              Cylinder {
                height 0.1
                radius 0.1
              }
            ]
          }
          Pose {
            translation -0.175 0 0
            rotation 1 0 0 1.5708
            children [
              Cylinder {
                height 0.1
                radius 0.1
              }
            ]
          }
          DEF Track_Box_Shape Box {
            size 0.35 0.1 0.2
          }
        ]
      }
      physics Physics {
      }
      device [
        LinearMotor {
          name "leftMotor"
        }
      ]
      animatedGeometry Shape {
        appearance Appearance {
          material Material {
            diffuseColor 0 0 0
          }
        }
        geometry DEF Track_Box_Shape Box {
          size 0.05 0.1 0.01
        }
      }
      geometriesCount 25
    }
    Track {
      translation 0 -0.3 -0.125
      children [
        TrackWheel {
          position 0.175 0
          rotation 0.36337976156201446 -0.6587685331909693 0.6587709530364497 2.444506309755461
          children [
            Shape {
              geometry Box {
                size 0.1 0.1 0.1
              }
            }
            DEF Sandwhich_Wheel Shape {
              appearance PBRAppearance {
              }
              geometry Cylinder {
                height 0.05
                radius 0.1
              }
            }
          ]
        }
        TrackWheel {
          position -0.175 0
          rotation 0.36337976156201446 -0.6587685331909693 0.6587709530364497 2.444506309755461
          children [
            Shape {
              geometry Box {
                size 0.1 0.1 0.1
              }
            }
            DEF Sandwhich_Wheel Shape {
              appearance PBRAppearance {
              }
              geometry Cylinder {
                height 0.05
                radius 0.1
              }
            }
          ]
        }
      ]
      name "track2"
      boundingObject Group {
        children [
          Pose {
            translation 0.175 0 0
            rotation 1 0 0 1.5708
            children [
              Cylinder {
                height 0.1
                radius 0.1
              }
            ]
          }
          Pose {
            translation -0.175 0 0
            rotation 1 0 0 1.5708
            children [
              Cylinder {
                height 0.1
                radius 0.1
              }
            ]
          }
          DEF Track_Box_Shape Box {
            size 0.35 0.1 0.2
          }
        ]
      }
      physics Physics {
      }
      device [
        LinearMotor {
          name "rightMotor"
        }
      ]
      animatedGeometry Shape {
        appearance Appearance {
          material Material {
            diffuseColor 0 0 0
          }
        }
        geometry DEF Track_Box_Shape Box {
          size 0.05 0.1 0.01
        }
      }
      geometriesCount 25
    }
    DEF Plate Shape {
      appearance BrushedAluminium {
        colorOverride 1 0.5 0.4
      }
      geometry Box {
        size 0.5 0.5 0.35
      }
    }
  ]
  boundingObject USE Plate
  physics Physics {
  }
  controller "BasicController"
}
Bench {
  translation 2.87 0 0
  rotation 0 0 1 2.61799
}
Bench {
  translation 2.39942 2.71111 0
  rotation 0 0 1 -1.3090053071795866
  name "bench(1)"
}
